Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/LoadWAND.py

```diff
@@ -147,7 +147,15 @@ def PyExec(self):
             if len(runs) == 1:
                 RenameWorkspace("__tmp_load", outWS, EnableLogging=False)
             else:
-                outName = outWS + "_" + str(mtd["__tmp_load"].getRunNumber())
+                try:
+                    temp_val = mtd["__tmp_load"].run().getTimeAveragedValue("HB2C:SE:SampleTemp")
+                except RuntimeError as _:
+                    temp_val = 300.0
+
+                if temp_val == 0.0:
```

Comment:
You may want to check for `nan` as well. I don't know if that would ever come out of the instrument.

Reply:
added it in